### ISSUE TITLE
final HYDE shows duplicate answer and html tags

### ISSUE THREAD

**Issue by slavag** [h2oai/h2ogpt#1408]
Hi,
When I'm running HYDE, started to get answer twice, first answer is in HTML tags and second is ok.

**Comment by pseudotensor** [comment]
Hi can you show example?  I can't tell what is meant exactly.

**Comment by slavag** [comment]
@pseudotensor please, copy & Paste from the answers box. 

Thanks

```
<details><summary><font size="3">HYDE 0: LLM</font></summary><font size="3"> Based on the information provided in the context, it appears that the site "helpdesk" in Tenant's SharePoint is showing up as a duplicate in Company Ltd, but it is actually a new site that needs to be backed up as it contains important client information.

The term "duplicate" in this context refers to a site that is being backed up in another backup task, specifically in Teams, to avoid data duplication and cost. The sites that have this status are mostly Team/Groups sites, and in some cases, if the SharePoint admin used for the SharePoint backup task has access to specific Team/Groups sites, Company Ltd's system auto-discovers them and adds them to the backup.

However, in this situation, the site "helpdesk" is not being backed up in Teams, but it is excluded from Teams backup. Therefore, it is important to inform Company Ltd's support team about this issue so they can update the backup configuration accordingly and remove the duplicate status for the site in SharePoint. This will ensure that the site is properly backed up and that all client information is protected.

To clarify, the duplicate status does not indicate that there is any actual duplicate data, but rather that the site is being backed up in another service, which in this case is Teams. Therefore, deleting the site with the duplicate status is not an option as there is no data to delete. Instead, Company Ltd's support team can help update the backup configuration to properly back up the site in SharePoint.

In summary, to answer the original question, it is important to inform Company Ltd's support team about the issue with the "helpdesk" site in SharePoint showing up as a duplicate but not being backed up in Teams, so they can update the backup configuration and remove the duplicate status to ensure proper backup of the site and its important client information.</font></details><details><summary><font size="3">HYDE 1: Prompt+LLM embedding</font></summary><font size="3"> Based on the information provided in the context, it appears that when a Team site is backed up in Microsoft Teams, it will be listed as a "Duplicate" in the SharePoint backup task. This is because both SharePoint and Teams backups are managed using one admin, and in some cases, if the SharePoint admin has access to specific Team sites, the system will auto-discover them and add them to the backup under Teams.

However, it's important to note that Company Ltd only backs up files once, and they cannot be stored twice in different tasks. If a user activates a site backup in a SharePoint task, they will not see the same site in a Teams/Groups task, and vice versa. If a user activates a site backup in a Teams/Groups task, the same site will be marked as "duplicated" in a SharePoint task, and it cannot be backed up again in SharePoint.

In the case of the "helpdesk" site in Tenant's SharePoint, since it's a new site, it's likely that it was not previously discovered by the system when the Teams backup was set up, but it was later discovered during the SharePoint backup. Therefore, it's showing up as a duplicate in the SharePoint backup task.

To address this issue, it's recommended to ensure that all necessary sites are added to the Teams backup task first, so they are not marked as duplicates in the SharePoint backup task. If the "helpdesk" site is important to back up, it should be added to the Teams backup task, and the duplicate status in SharePoint should no longer be an issue.

Additionally, it's important to note that the "Duplicate" status is just a label to indicate that the site is already being backed up under another task, and it does not mean that there is actually any data duplication occurring. The data is still stored only once, and the label is just to help users manage their backups more effectively.</font></details><br> Based on the information provided in the context, it appears that the "Duplicate" status in Company Ltd for the SharePoint site "helpdesk" is due to the fact that it is also being backed up under the Teams or Groups backup task. Since both SharePoint and Teams or Groups backup tasks are being managed using one admin account, the system auto-discovers and adds Team/Groups sites to the SharePoint backup task to avoid data duplication.

Therefore, it is not possible to clear the "Duplicate" status for the "helpdesk" site in the SharePoint backup task as it is already being backed up under the Teams or Groups backup task. However, the site's data is being backed up, and it is accessible through the Teams or Groups recovery list.

If it is important to have the "helpdesk" site backed up under the SharePoint backup task for any reason, it would be necessary to create a new SharePoint backup task or manually add the site to the existing SharePoint backup task to ensure that it is being backed up separately from the Teams or Groups backup task.

I hope this information is helpful. Let me know if you have any further questions or concerns.
```

**Comment by pseudotensor** [comment]
What I see:

![image](https://github.com/h2oai/h2ogpt/assets/2249614/a9f90945-d17c-47a8-b67b-0694db014776)


I don't see repetition.

**Comment by slavag** [comment]
@pseudotensor but inside answer I don't see any open/close arrows, I see the html tags, and btw "Only final HYDE shown" is marked

**Comment by pseudotensor** [comment]
Sorry, I am having trouble understanding what you mean.

The standard view I see is:

![image](https://github.com/h2oai/h2ogpt/assets/2249614/d5359a32-a141-4c7c-b9a4-4fd476484c0e)

Then I can click on the triangle to open up the accordion to see what I shared above.

By "inside the answer I don't see any open/close arrows" what do you mean?  By arrows you mean the accordion triangle thing?  But the code you shared shows them.  Are you saying it's not rendered properly in the gradio UI?  If so  please share.

I don't know what '"Only final HYDE shown" is marked' means, please explain.

**Comment by slavag** [comment]
@pseudotensor this is how it's looks like
<img width="1145" alt="image" src="https://github.com/h2oai/h2ogpt/assets/511540/79446db1-a6cc-4008-8781-517dc5a7b1bd">

**Comment by pseudotensor** [comment]
Ah ok, so it's not rendering properly by gradio UI.  Did this start after upgrading to some new h2oGPT that had some new gradio?

**Comment by pseudotensor** [comment]
https://github.com/gradio-app/gradio/issues/7198

**Comment by slavag** [comment]
@pseudotensor Thanks
